[CI] EsqlSpecIT test {lookup-join.MvJoinKeyFromRow ASYNC} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #5852 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/7dr43766u73uu)
- [elasticsearch-pull-request #50194 / part-3](https://gradle-enterprise.elastic.co/s/2clqry6xpt7t4)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT" -Dtests.method="test {lookup-join.MvJoinKeyFromRow SYNC}" -Dtests.seed=C8A7015F871BACD4 -Dtests.locale=sw-UG -Dtests.timezone=America/Catamarca -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Blookup-join.MvJoinKeyFromRow%20ASYNC%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Data mismatch:
row 0 column 1:a list containing
row 0 column 1:0: expected "Quenya" but was "Mv-Lang2"
row 0 column 2:a list containing
row 0 column 2:0: expected "null" but was "Mv-Land2"
row 1 column 1:a list containing
row 1 column 1:0: expected "null" but was "Quenya"
row 1 column 2:a list containing
row 1 column 2:0: expected "Atlantis" but was "null"
row 2 column 1:a list containing
row 2 column 1:0: expected "Mv-Lang" but was "null"
row 2 column 2:a list containing
row 2 column 2:0: expected "Mv-Land" but was "Atlantis"
row 3 column 1:a list containing
row 3 column 1:0: expected "Mv-Lang2" but was "Mv-Lang"
row 3 column 2:a list containing
row 3 column 2:0: expected "Mv-Land2" but was "Mv-Land"
Actual:
language_code:integer | language_name:keyword | country:text
[4, 5, 6, 7]          | Mv-Lang2              | Mv-Land2    
[4, 5, 6, 7]          | Quenya                | null        
[4, 5, 6, 7]          | null                  | Atlantis    
[4, 5, 6, 
[truncated]
```

**Issue Reasons:**
- [main] 2 failures in test test {lookup-join.MvJoinKeyFromRow ASYNC} (0.4% fail rate in 495 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -251,9 +251,6 @@ tests:
 - class: org.elasticsearch.oldrepos.OldRepositoryAccessIT
   method: testOldSourceOnlyRepoAccess
   issue: https://github.com/elastic/elasticsearch/issues/120080
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {lookup-join.MvJoinKeyFromRow ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/120242
 
 # Examples:
 #

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join.csv-spec
@@ -450,13 +450,14 @@ required_capability: join_lookup_v11
 ROW language_code = [4, 5, 6, 7]
 | LOOKUP JOIN languages_lookup_non_unique_key ON language_code
 | KEEP language_code, language_name, country
+| SORT language_code, language_name, country
 ;
 
 language_code:integer | language_name:keyword | country:text
-[4, 5, 6, 7]          | Quenya                | null
-[4, 5, 6, 7]          | null                  | Atlantis
 [4, 5, 6, 7]          | Mv-Lang               | Mv-Land
 [4, 5, 6, 7]          | Mv-Lang2              | Mv-Land2
+[4, 5, 6, 7]          | Quenya                | null
+[4, 5, 6, 7]          | null                  | Atlantis
 ;
 
 mvJoinKeyFromRowExpanded

